### ISSUE TITLE
Add integration for `graphql-type-json` to API category

### ISSUE THREAD

**Issue by jkeys-ecg-nmsu** [aws-amplify/amplify-cli#555]
**Is your feature request related to a problem? Please describe.**
We have third-party data providers sending us JSON blobs that represent discrete records. It would be nice to have native integration for the npm package `graphql-type-json`.  

https://www.npmjs.com/package/graphql-type-json

![image](https://user-images.githubusercontent.com/22964702/49601074-0d3c1800-f942-11e8-80ee-87fca4d803a5.png)

**Describe the solution you'd like**
Native integration with `graphql-type-json` for the API category, including code generation. Currently this is what happens when attempting to add scalar JSON to the GraphQL schema. (I had planned on adding the relevant resolver to the generated client-side code.)

**Describe alternatives you've considered**
Use the String scalar and use `JSON.stringify()` to transform the parsed Javascript objects into JSON, and JSON.parse() to cast the String to an Object.

**Additional context**
The CLI claims that the Schema compilation is successful; the error happens at the CloudFormation build process.

Re whether this is worth supporting: Despite not being updated in eight months and having a version of 0.x, that package has over 161k downloads in the past week, so I believe it is basically stable.

**Comment by jkeys-ecg-nmsu** [comment]
I think this Issue might actually be invalid or unnecessary. I found in the AppSync documentation that AppSync provides its own `AWSJSON` scalar. When I updated my schema to use this, `amplify push` succeeded and the generated code makes use of AWSJSON.

https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html

Are all AWS-defined GraphQL scalars supported by Amplify CLI? If so, feel free to close this issue.
